homeassistant/components/satel_integra.py

```diff
@@ -0,0 +1,150 @@
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICE_HOST): cv.string,
+        vol.Optional(CONF_DEVICE_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_DEVICE_PARTITION,
+                     default=DEFAULT_DEVICE_PARTITION): cv.positive_int,
+        vol.Optional(CONF_ARM_HOME_MODE,
+                     default=DEFAULT_CONF_ARM_HOME_MODE): vol.In([1, 2, 3]),
+        vol.Optional(CONF_ZONES): {vol.Coerce(int): ZONE_SCHEMA},
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the Satel Integra component."""
+    conf = config.get(DOMAIN)
+
+    zones = conf.get(CONF_ZONES)
+    host = conf.get(CONF_DEVICE_HOST)
+    port = conf.get(CONF_DEVICE_PORT)
+    partition = conf.get(CONF_DEVICE_PARTITION)
+
+    from satel_integra.satel_integra import AsyncSatel, AlarmState
+
+    controller = AsyncSatel(host, port, zones, hass.loop, partition)
+
+    hass.data[DATA_SATEL] = controller
+
+    result = yield from controller.connect()
+
+    if not result:
+        return False
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP,
+                               lambda s: controller.close())
+
+    _LOGGER.debug("Arm home config: %s, mode: %s ",
+                  conf,
+                  conf.get(CONF_ARM_HOME_MODE))
+
+    yield from async_load_platform(hass, 'alarm_control_panel',
+                                   DOMAIN, conf, config)
+
+    yield from async_load_platform(hass, 'binary_sensor',
+                                   DOMAIN, {CONF_ZONES: zones}, config)
+
+    @callback
+    def alarm_status_update_callback(status):
+        """Send status update received from alarm to home assistant."""
+        _LOGGER.debug("Alarm status callback, status: %s", status)
+        hass_alarm_status = STATE_ALARM_DISARMED
+
+        if status == AlarmState.ARMED_MODE0:
+            hass_alarm_status = STATE_ALARM_ARMED_AWAY
+
+        elif status in [
+                AlarmState.ARMED_MODE0,
+                AlarmState.ARMED_MODE1,
+                AlarmState.ARMED_MODE2,
+                AlarmState.ARMED_MODE3
+        ]:
+            hass_alarm_status = STATE_ALARM_ARMED_HOME
+
+        elif status in [AlarmState.TRIGGERED, AlarmState.TRIGGERED_FIRE]:
+            hass_alarm_status = STATE_ALARM_TRIGGERED
+
+        elif status == AlarmState.DISARMED:
+            hass_alarm_status = STATE_ALARM_DISARMED
+
+        _LOGGER.debug("Sending hass_alarm_status: %s...", hass_alarm_status)
+        async_dispatcher_send(hass, SIGNAL_PANEL_MESSAGE, hass_alarm_status)
+
+    @callback
+    def zones_update_callback(status):
+        """Update zone objects as per notification from the alarm."""
+        _LOGGER.debug("Zones callback , status: %s", status)
+        async_dispatcher_send(hass, SIGNAL_ZONES_UPDATED, status[ZONES])
+
+    hass.async_add_job(asyncio_ensure_future(controller.keep_alive()))
```

Comment:
don't do that. Run only a job, you can wrap this if that is no coro

Reply:
Can you give me some more guidance here? 
If I simply remove "asyncio_ensure_future", the code blocks, the `async_setup_platform` never quits. 
It's really my first contact with asyncio so I must be missing something. `keep_alive` is a coro.

Reply:
@pvizeli can you please advice?

Reply:
OK, wrapped in `hass.loop.create_task`.